### PR TITLE
Feature/rating catalog

### PR DESCRIPTION
**Pull request by rripken**
Created end-points for RatingTemplate and RatingSpec

### CODE REVIEW

File: cwms_radar_api/src/main/java/cwms/radar/api/RatingSpecController.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            @OpenApiParam(name="office", description="Specifies the owning office of the Rating Specs whose data is to be included in the response. If this field is not specified, matching rating information from all offices shall be returned."),
+            @OpenApiParam(name="rating-id-mask", description="RegExp that specifies the rating IDs to be included in the response. If this field is not specified, all Rating Specs shall be returned."),
+                @OpenApiParam(name="page",
+                        description = "This end point can return a lot of data, this identifies where in the request you are. This is an opaque value, and can be obtained from the 'next-page' value in the response."
+                ),
+                @OpenApiParam(name="page-size", type=Integer.class,
+                        description = "How many entries per page returned. Default " + DEFAULT_PAGE_SIZE + "."
+                ),
+        },
+            responses = {
```
Comment (DanielTOsborne) [inline cwms_radar_api/src/main/java/cwms/radar/api/RatingSpecController.java]: Indentation level doesn't match the surrounding items.
[Line 116](https://github.com/USACE/cwms-radar-api/pull/164/files#diff-b486a96bd450422a047bbf500f6060f007d6052277965934308356e397d951c2R116) is a good reference.

Nitpick, feel free to ignore. :-)

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingSetDao.java [hunk 2]
```diff
@@ -92,30 +118,29 @@ public void delete(String officeId, String ratingSpecId) throws IOException, Rat
 	{
 		try
 		{
-			dsl.connection(c ->
-			{
-//				deleteWithRatingSet(c, officeId, ratingSpecId);
+			dsl.connection(c -> {
+				//				deleteWithRatingSet(c, officeId, ratingSpecId);
```
Comment (DanielTOsborne) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingSetDao.java]: This comment got further indented with the formatting change.
Is there a reason this commented out code even needs to remain?

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/rating/RatingSpec.java [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+		return effectiveDates;
+	}
+
+	@Override
+	public void validate() throws FieldException
+	{
+
+	}
+
+	public static class IndRoundingSpec {
```
Comment (DanielTOsborne) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/rating/RatingSpec.java]: I feel if the full name is spelled out in [RatingTemplate.java](https://github.com/USACE/cwms-radar-api/pull/164/files#diff-5f04b8de36b6b1f611173776126c3d052b2d15f689420e6339ff0b9a82ec7e46) then it probably should be here as well, just for sake of consistency. So `IndependentRoundingSpec`, for example.

Same for the properties in parent class (e.g. `getIndependentRoundingSpecs()`, `getDependentRoundingSpecs()`).

File: cwms_radar_api/src/main/java/cwms/radar/formatters/json/JsonV1.java [hunk 4]
```diff
@@ -42,7 +46,8 @@
 				   Clob.class,
 				   Clobs.class,
 				   TimeSeriesGroup.class,
-				   RecentValue.class
+				   RecentValue.class,
+				   RatingTemplate.class, RatingTemplates.class, RatingSpec.class, RatingSpecs.class
```
Comment (DanielTOsborne) [inline cwms_radar_api/src/main/java/cwms/radar/formatters/json/JsonV1.java]: Same as the comment in #165 should be JSONv2.

File: cwms_radar_api/src/main/java/cwms/radar/api/RatingTemplateController.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            @OpenApiParam(name="template-id-mask", description="RegExp that specifies the rating template IDs to be included in the response. If this field is not specified, all rating templates shall be returned."),
+                @OpenApiParam(name="page",
+                        description = "This end point can return a lot of data, this identifies where in the request you are. This is an opaque value, and can be obtained from the 'next-page' value in the response."
+                ),
+                @OpenApiParam(name="page-size",
+                        type=Integer.class,
+                        description = "How many entries per page returned. Default " + DEFAULT_PAGE_SIZE + "."
+                ),
+        },
+            responses = {
```
Comment (DanielTOsborne) [inline cwms_radar_api/src/main/java/cwms/radar/api/RatingTemplateController.java]: Indentation here as well, if you're going to fix them.
I also just noticed here, and in the RatingSPecController.java file, that the OpenApiParam annotations above are also not aligned. I thought they were nested at first, but they don't appear to be (line 57 in this file).

File: cwms_radar_api/src/main/java/cwms/radar/api/RatingSpecController.java (hunk 1 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/api/RatingSpecController.java]: Will fix

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingSetDao.java (hunk 2 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingSetDao.java]: Nope. Will delete.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/rating/RatingSpec.java (hunk 3 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/rating/RatingSpec.java]: Agreed

File: cwms_radar_api/src/main/java/cwms/radar/formatters/json/JsonV1.java (hunk 4 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/formatters/json/JsonV1.java]: Yep.

File: cwms_radar_api/src/main/java/cwms/radar/api/RatingTemplateController.java (hunk 5 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/api/RatingTemplateController.java]: Yeah, weird.  Will fix.

**Review by DanielTOsborne — DISMISSED** [review]

**Review by DanielTOsborne — APPROVED** [review]

**Pull request merged**